File: Tema 1/Error6.cpp
#include <iostream>
#include <limits>

int main() {
    int max = std::numeric_limits<int>::max(); // 2,147,483,647
    int resultado = max + 1; // Desbordamiento

    std::cout << "Maximo int: " << max << std::endl;
    std::cout << "Maximo + 1: " << resultado << std::endl;

    // Detección manual del desbordamiento (estilo didáctico)
    if (resultado < max) {
        std::cout << "Error detectado: ¡Desbordamiento!" << std::endl;
    }

    return 0;
}

File: Tema 1/Error6.java
public class Error6 {
    public static void main(String[] args) {
        int max = Integer.MAX_VALUE; // 2,147,483,647
        int resultado = max + 1;

        System.out.println("Máximo int: " + max);
        System.out.println("Máximo + 1: " + resultado); // Imprime -2147483648

        // Solución: Usar Math.addExact para lanzar una excepción si ocurre
        try {
            Math.addExact(max, 1);
        } catch (ArithmeticException e) {
            System.out.println("Error detectado: ¡Desbordamiento!");
        }
    }
}